app/Http/Controllers/BookController.php

```diff
@@ -3,34 +3,123 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
-use App\Http\Requests\BookRequest;
-use App\Policies\BookPolicy;
 use App\Http\Models\Book;
-use App\Http\Models\User;
-use App\Http\Models\Category;
-use App\Http\Models\Publisher;
+use App\Repositories\Book\BookRepoInterface;
+use App\Repositories\Category\CategoryRepoInterface;
+use App\Repositories\Publisher\PublisherRepoInterface;
+use App\Repositories\User\UserRepoInterface;
+use App\Http\Requests\BookRequest;
 use Illuminate\Support\Facades\Auth;
-use App\Enums\UserRole;
-use Illuminate\Support\Str;
 
 class BookController extends Controller
 {
-    public function index()
+    protected $bookRepo;
+    protected $categoryRepo;
+    protected $pubRepo;
+    protected $userRepo;
+
+    function __construct(BookRepoInterface $bookRepo, CategoryRepoInterface $categoryRepo,
+        PublisherRepoInterface $pubRepo,UserRepoInterface $userRepo)
```

Comment:
xuống dòng thì xuống hết đi
```suggestion
    function __construct(
        BookRepoInterface $bookRepo,
        CategoryRepoInterface $categoryRepo,
        PublisherRepoInterface $pubRepo,
        UserRepoInterface $userRepo
    ) {
```

Reply:
vang a